core/application/impl/app_state_manager_impl.hpp

```diff
@@ -52,7 +52,7 @@ namespace kagome::application {
 
     log::Logger logger_;
 
-    State state_ = State::Init;
+    std::atomic<State> state_ = State::Init;
```

Comment:
Is it actually needed or just left here after trying to fix TSAN?

Reply:
Not sure. It did not help for tsan, but can be useful for use app_state_manager in multithreading environment.